astropy/io/ascii/ecsv.py

```diff
@@ -124,6 +125,13 @@ def get_cols(self, lines):
 
         try:
             header = meta.get_header_from_yaml(lines)
+        except ImportError as exc:
+            if 'PyYAML package is required' in str(exc):
+                warnings.warn("file looks like ECSV format but PyYAML is not installed "
+                              "so it cannot be parsed as ECSV",
+                              AstropyWarning)
+            raise core.InconsistentTableError('unable to parse yaml in meta header'
```

Comment:
I would have preferred to have this raise the `ImportError`, but that doesn't get caught in guessing.  Hence the compromise of `InconsistentTableError` with text about the import.